src/server/main.cpp

```diff
@@ -14,16 +14,54 @@
 
 #include <memory>
 #include <thread>
+#include <queue> 
+#include <mutex> 
+#include <condition_variable>
 
 #include <iostream>
 namespace
 {
 
 std::unique_ptr<alicia::LoginDirector> g_loginDirector;
 std::unique_ptr<alicia::RanchDirector> g_ranchDirector;
+std::queue<std::function<void()>> g_taskQueue; // Task queue to hold lambdas.
```

Comment:
You already did the `Scheduler` class, move its implementation to a new `Scheduler.cpp` file instead of it being here.

Reply:
Oh nvm, I just noticed that martinprejsa made the change